tests/test_automatic_fields.py

```diff
@@ -47,10 +54,18 @@ def test_run(self):
                     msg="The number of records is not over the stream max limit")
 
                 # verify that only the automatic fields are sent to the target
-                expected_fields_for_stream = (self.expected_primary_keys().get(stream, set()) |
+                expected_fields_for_stream = (expected_primary_keys |
                                               self.top_level_replication_key_fields().get(stream, set()) |
                                               self.expected_foreign_keys().get(stream, set()))
                 self.assertEqual(
                     actual_fields_by_stream.get(stream, set()),
                     expected_fields_for_stream,
                     msg="The fields sent to the target are not the automatic fields.\nExpected: {}\nActual: {}".format(expected_fields_for_stream, actual_fields_by_stream.get(stream, set())))
+
+                # Verify that all replicated records have unique primary key values
+                records_pks_set = {tuple([message.get('data').get(primary_key) for primary_key in expected_primary_keys])
+                                          for message in messages.get('messages')}
+                records_pks_list = [tuple([message.get('data').get(primary_key) for primary_key in expected_primary_keys])
+                                           for message in messages.get('messages')]
+                self.assertCountEqual(records_pks_set, records_pks_list,
+                                      msg="We have duplicate records for {}".format(stream))
```

Comment:
Improve readability of code as discussed by removing repeated code.

Reply:
Removed duplicate code.